File: protocol/src/test/csharp/CsProtocol/Packet/ObjectB.cs
using System;
using System.Collections.Generic;
using CsProtocol.Buffer;

namespace CsProtocol
{
    // @author godotg
    // @version 3.0
    public class ObjectB : IPacket
    {
        public bool flag;

        public static ObjectB ValueOf(bool flag)
        {
            var packet = new ObjectB();
            packet.flag = flag;
            return packet;
        }


        public short ProtocolId()
        {
            return 103;
        }
    }


    public class ObjectBRegistration : IProtocolRegistration
    {
        public short ProtocolId()
        {
            return 103;
        }

        public void Write(ByteBuffer buffer, IPacket packet)
        {
            if (buffer.WritePacketFlag(packet))
            {
                return;
            }
            ObjectB message = (ObjectB) packet;
            buffer.WriteBool(message.flag);
        }

        public IPacket Read(ByteBuffer buffer)
        {
            if (!buffer.ReadBool())
            {
                return null;
            }
            ObjectB packet = new ObjectB();
            bool result0 = buffer.ReadBool();
            packet.flag = result0;
            return packet;
        }
    }
}

File: protocol/src/test/go/protocol/ObjectB.go
package protocol

type ObjectB struct {
	flag bool
}

func (protocol ObjectB) protocolId() int16 {
	return 103
}

func (protocol ObjectB) write(buffer *ByteBuffer, packet any) {
	if buffer.WritePacketFlag(packet) {
		return
	}
	var message = packet.(*ObjectB)
	buffer.WriteBool(message.flag)
}

func (protocol ObjectB) read(buffer *ByteBuffer) any {
	var packet = new(ObjectB)
	if !buffer.ReadBool() {
		return packet
	}
	var result0 = buffer.ReadBool()
	packet.flag = result0
	return packet
}
